src/kakarot/accounts/account_contract.cairo

```diff
@@ -133,19 +137,37 @@ func __execute__{
         assert call_array_len = 1;
     }
 
-    let latest_class = AccountContract.get_latest_class();
+    let (latest_account_class, latest_cairo1helpers_class) = AccountContract.get_latest_classes();
     let (this_class) = Account_implementation.read();
-    if (latest_class != this_class) {
-        // Must be done before library_call, otherwise entering an infinite recursive loop.
-        Account_implementation.write(latest_class);
+    if (latest_account_class != this_class) {
+        // Update storage
+        Account_implementation.write(latest_account_class);
+        let (kakarot_address) = Ownable_owner.read();
+        let (this_cairo1helpers_class) = Account_cairo1_helpers_class_hash.read();
+
+        if (latest_cairo1helpers_class != this_cairo1helpers_class) {
```

Comment:
what if we change the helper but not the account class? I think that they need to be in separate branches, and that overall the function can be simplified like

```
let (account_class, helpers_class) = AccountContract.get_latest_classes();
let (this_helpers_class) = Account_helpers_class_hash.read();

if (helpers_class != this_cairo1helpers_class) {
    // Note this could be done in any cases but would cost a useless write
    Account_cairo1_helpers_class_hash.write(latest_cairo1helpers_class);
}

// Do the call with the latest account class, possibly same class, works in all the cases
let (response_len, response) = IAccount.library_call___execute__(
    class_hash=latest_account_class,
    call_array_len=call_array_len,
    call_array=call_array,
    calldata_len=calldata_len,
    calldata=calldata,
);
```

we don't need to actually store the implementation if we always call kakarot to actually check what is the implementation.